Remove nodemon from dev server lifecycle (#4407)
wasp-lang/wasp: waspc/ChangeLog.md

### 🐞 Bug fixes

- Fixed a race condition where `useAuth()` could return stale user data after an action that modifies the `User` entity (by @okxint). ([#4343](https://github.com/wasp-lang/wasp/issues/4343))

> - `wasp start` now manages the development server process directly instead of relying on the generated server's `nodemon`, improving restart behavior. ([#4407](https://github.com/wasp-lang/wasp/pull/4407))

Reviewer: nitpick

```suggestion
- `wasp start` now manages the development server process directly instead with `nodemon`, improving stability. ([#4407](https://github.com/wasp-lang/wasp/pull/4407))
```
